mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -218,15 +216,11 @@ private boolean hadCID(Entity part) {
         return readHeader(part, CONTENT_ID, ContentIdField.class).isPresent();
     }
 
-    private InputStream getContent(Body body) throws IOException {
-        if (body instanceof SingleBody) {
-            SingleBody singleBody = (SingleBody) body;
-            return singleBody.getInputStream();
-        }
+    private byte[] getContent(Body body) throws IOException {
         DefaultMessageWriter messageWriter = new DefaultMessageWriter();
         ByteArrayOutputStream out = new ByteArrayOutputStream();
         messageWriter.writeBody(body, out);
-        return new ByteArrayInputStream(out.toByteArray());
+        return out.toByteArray();
```

Comment:
oh, anyway it was already a byteArray, okish. But why this change then?

Reply:
Because @mbaechler was not happy with the cast and byte array reuse.

Keeping the copy seemed more consensual, hence I executed myself :-)